ui/src/containers/NodeVolumes.js

```diff
@@ -31,44 +42,109 @@ const VolumeTable = styled.div`
   margin-top: ${padding.small};
 `;
 
+const DeleteConfirmationModal = {
+  content: {
+    top: '50%',
+    left: '50%',
+    width: '540px',
+    height: '190px',
+    transform: 'translate(-50%, -50%)'
+  }
+};
+
+const NotificationButtonGroup = styled.div`
+  display: flex;
+  justify-content: center;
+  margin-top: ${padding.larger};
+  margin-left: 270px;
+`;
+
+const DeleteButton = styled(Button)`
+  margin-left: ${padding.small};
+`;
+
 const NodeVolumes = props => {
+  const { intl } = props;
+
   useRefreshEffect(refreshVolumesAction, stopRefreshVolumesAction);
   useRefreshEffect(
     refreshPersistentVolumesAction,
     stopRefreshPersistentVolumesAction
   );
+
   const volumes = useSelector(state => state.app.volumes);
+  const persistentVolumes = useSelector(state => state.app.volumes.pVList);
   const [sortBy, setSortBy] = useState('name');
   const [sortDirection, setSortDirection] = useState('ASC');
+  const [
+    isDeleteConfirmationModalOpen,
+    setisDeleteConfirmationModalOpen
+  ] = useState(false);
+  const [deleteVolumeName, setDeleteVolumeName] = useState('');
   const onSort = ({ sortBy, sortDirection }) => {
     setSortBy(sortBy);
     setSortDirection(sortDirection);
   };
 
+  const isVolumeDeletable = rowData => {
+    const volumeStatus = rowData.status;
+    const volumeName = rowData.name;
+
+    if (
```

Comment:
Why not using switch here?
Would be more readable than `if/else`.

This kind of control flow (branching on a finite set of value) is one of the best use case for switch.

e.g.:

```javascript
switch (volumeStatus) {
  case STATUS_VOLUME_UNKNOWN:
  case STATUS_PENDING:
  case STATUS_TERMINATING:
    return false;
  case STATUS_FAILED:
  case STATUS_AVAILABLE: {
    const persistentVolume = persistentVolumes.find(
      pv => pv?.metadata?.name === volumeName
    );
    const persistentVolumeStatus = persistentVolume?.status?.phase;
    switch (persistentVolumeStatus) {
      case STATUS_FAILED:
      case STATUS_AVAILABLE:
      case STATUS_RELEASED:
        return true;
      case STATUS_PENDING:
      case STATUS_BOUND:
        return false;
      default:
        // Log a warning/error here, not supposed to happen.
        return false;
    }
  }
  default:
    // Log a warning/error here, not supposed to happen.
    return false;
}
```